nitro-root-enclave/src/main.rs

```diff
@@ -0,0 +1,467 @@
+//! AWS Nitro-Enclaves-specific material for the Root Enclave
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+use lazy_static::lazy_static;
+use nitro_enclave_token::{AttestationDocument, NitroToken};
+use nix::sys::socket::listen as listen_vsock;
+use nix::sys::socket::{accept, bind, SockAddr};
+use nix::sys::socket::{socket, AddressFamily, SockFlag, SockType};
+use psa_attestation::{psa_initial_attest_get_token, psa_initial_attest_load_key, t_cose_sign1_get_verification_pubkey};
+use ring;
+use ring::signature::KeyPair;
+use ring::signature::{EcdsaKeyPair, ECDSA_P256_SHA256_FIXED_SIGNING};
+use std::sync::Mutex;
+use veracruz_utils::{NitroRootEnclaveMessage, NitroStatus};
+
+use veracruz_utils::{receive_buffer, send_buffer};
+
+use nsm_io;
+use nsm_lib;
+
+//const CID: u32 = 17;
+const CID: u32 = 0xFFFFFFFF; // VMADDR_CID_ANY
+const PORT: u32 = 5005;
+// Maximum number of outstanding connections in the socket's
+// listen queue
+const BACKLOG: usize = 128;
+
+static AWS_NITRO_ROOT_CERTIFICATE: [u8; 533] = [
```

Comment:
Is this root certificate ever going to change?  I wonder if factoring the contents of the file out and using `include!()` (see: https://doc.rust-lang.org/std/macro.include.html) is a good idea?

Reply:
According to AWS, it should never change (and the certificate's expiry is in 2049). If they do need to change it, something horrible has happened at AWS.